packages/typespec-autorest/test/models.test.ts

```diff
@@ -50,13 +47,43 @@ describe("typespec-autorest: model definitions", () => {
       };`
     );
 
-    ok(res.isRef);
-    deepStrictEqual(res.defs.Foo, {
-      type: "object",
+    expect(res.defs.Foo).toMatchObject({
       properties: {
         xJson: { type: "integer", format: "int32", "x-ms-client-name": "x" },
       },
-      required: ["xJson"],
+    });
+  });
+
+  it("uses json name specified via @encodedName", async () => {
+    const res = await oapiForModel(
+      "Foo",
+      `model Foo {
+        @encodedName("application/json", "xJson")
+        x: int32;
+      };`
+    );
+
+    expect(res.defs.Foo).toMatchObject({
+      properties: {
+        xJson: { type: "integer", format: "int32" },
```

Comment:
shouldn't we be setting x-ms-client-name to match the original name here?

Reply:
let me add it I think this might just be the `toMatchObject` which doesn't care if there is extra properties

Reply:
Add to the test